### DISCUSSION TITLE
How to integrate skywalking with istio to support mesh tracing？

### DISCUSSION THREAD

**Discussion by whl12345** [apache/skywalking#8399] [Q&A]
I have known that skywalking has supported the integration with istio including metric and topology not tracing.But we expect skywalking to support tracing like zipkin in istio especially for envoy span.With skywalking agent for java , we can get tracing, however envoy span will be lot. We could make our tracing more integrated when envoy span is included.
I found envoy/istio 1.11 has supported skywalking, how can we use it to add envoy span in tracing?

**Comment by wu-sheng** [comment]
Don't know what do you mean. Sorry.
What is the point of Zipkin here? SkyWalking is in the Istio supported list for a long time.

**Reply by whl12345** [reply]
Istio has supported zipkin tracer and we can get more integrated trace. 
Example for service A request service B in istio and we should get a trace like that
**A(http client span)->envoy outbound span-> envoy inbound span-> B(tomcat span)**
With skywalking agent, we can only get a trace like **A(http client span)->B(tomcat span)**  without envoy span.To achieve it,I think It need envoy proxy to support v3 propagation protocol in skywalking like b3 protocol in zipkin.

**Reply by wu-sheng** [reply]
It is there for a long time

https://github.com/istio/istio/pull/32588

Maybe Istio team changed the doc unexpectedly somehow. I am not following day by day.

**Reply by whl12345** [reply]
Thanks a lot, I found the PR.
Is there a detailed official document or demo to help us integrate skywalking tracer with istio？
Just run "istioctl dashboard skywalking" without any other config?

**Reply by wu-sheng** [reply]
To @whl12345 I think https://github.com/istio/istio/pull/32588#issuecomment-862002263 is the right context to follow. It is easy to change from Zipkin to SkyWalking. Not much config to add.

@Shikugawa maybe know more.

**Reply by berylshow** [reply]
> @wu-sheng 
Hello, can istio's telemetry API be used with skywalking agent? Will the metrics data double?

**Reply by wu-sheng** [reply]
Yes, it will. If you want both, use different service name formats.

**Reply by berylshow** [reply]
> Yes, it will. If you want both, use different service name formats.

@wu-sheng  OK, I see. Thank you~
